### PR TITLE
[WIP] update to SBO library to 1.0

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**

/kind dependency-change

**What does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/5254
fixes https://github.com/openshift/odo/issues/5207

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by feloy** [comment]
Kubernetes Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-87.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-87.html)

**Comment by feloy** [comment]
Unit Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-100.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-100.html)

**Comment by feloy** [comment]
OpenShift Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-116.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-116.html)

**Comment by feloy** [comment]
> OpenShift Tests finished. View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-116.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-116.html)

The errors with `Bailing out early after 300s` are mine, I still haven't lost the habit to test on the OpenShift cluster :(

**Comment by feloy** [comment]
Kubernetes Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-100.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-100.html)

**Comment by feloy** [comment]
OpenShift Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-129.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-129.html)

**Comment by feloy** [comment]
Unit Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-113.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-113.html)

**Comment by feloy** [comment]
Kubernetes Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-102.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-kubernetes-tests-102.html)

**Comment by feloy** [comment]
OpenShift Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-131.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-openshift-tests-131.html)

**Comment by feloy** [comment]
Unit Tests finished.
View logs: [TXT](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-115.txt) [HTML](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-5255-unit-tests-115.html)

File: go.mod [hunk 1]
```diff
@@ -71,6 +71,7 @@ replace (
 	github.com/containers/image => github.com/openshift/containers-image v0.0.0-20190130162819-76de87591e9d
 	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20191121165722-d1d5f6476656
 	github.com/onsi/ginkgo => github.com/openshift/onsi-ginkgo v4.7.0-origin.0+incompatible
+	github.com/redhat-developer/service-binding-operator => github.com/kadel/service-binding-operator v0.4.1-0.20211207095808-c49634ebeb6f
```
Comment (kadel) [inline go.mod:74]: TODO (@kadel): his was done to verify fix for panic. Fix was merged upstream (https://github.com/redhat-developer/service-binding-operator/pull/1079) Now this is not needed, and we should be able to  use upstream service-biding-operator

Comment (kadel) [reply go.mod]: done

**Comment by kadel** [comment]
/close
in favor of https://github.com/redhat-developer/odo/pull/5314

**Pull request closed without merging**